include/swift/SILOptimizer/Utils/BasicBlockOptUtils.h

```diff
@@ -33,6 +33,24 @@ class BasicBlockCloner;
 class SILLoop;
 class SILLoopInfo;
 
+/// Compute the set of reachable blocks.
+class ReachableBlocks {
+  llvm::SmallPtrSet<SILBasicBlock *, 32> visited;
+
+public:
+  /// Invoke \p visitor for each reachable block in \p f in worklist order (at
+  /// least one predecessor has been visited--defs are always visited before
+  /// uses except for phi-type block args). The \p visitor takes a block
+  /// argument, which is already marked visited, and must return true to
+  /// continue visiting blocks.
+  ///
+  /// Returns true if all reachable blocks were visited.
+  bool visit(SILFunction *f, std::function<bool(SILBasicBlock *)> visitor);
```

Comment:
Shouldn't this be an function_ref? You aren't going to ever store/escape the std::function. I guess it is better to be conservative though.

Reply:
It can be `function_ref`, I didn't think carefully enough about it.